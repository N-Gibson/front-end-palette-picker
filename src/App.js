import React, { Component } from 'react';
import Projects from './Components/Projects/projects';
import { Route } from 'react-router-dom';
import './App.css';
import Palette from './Components/Palette/palette'
import AllPalettes from './Components/AllPalettes/AllPalettes'

class App extends Component {
  constructor() {
    super()
    this.state = {
      error: '',
      paletteInfo: []
    }
  }

  handleInfo = (palettes) => {
    this.setState({ paletteInfo: palettes })
  }

  render() {
    return (
      <div className="App">
      <Route exact path='/' render={() => <Projects />}/>
      <Route exact path='/project/:id' render={({ match }) => {
        const { id } = match.params;
        return <Palette handleInfo={this.handleInfo} id={id}/>
      }} />
      <Route exact path='/palettes/:id' render={() => { 
        return <AllPalettes palettes={this.state.paletteInfo} />
      }}/>
      </div>
    );
  }
}

export default App;
