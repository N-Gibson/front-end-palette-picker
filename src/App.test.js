import React from 'react';
import { shallow } from 'enzyme';
import App from './App'

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);

    wrapper.setState({
      error: '',
      paletteInfo: [],
      currentProject: ''
    })
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update the palette information', () => {
    expect(wrapper.state('paletteInfo')).toEqual([])
    wrapper.instance().handleInfo([
      {hex: '#fff', name: 'dog'},
      {hex: '#000', name: 'bob'},
      {hex: '#fff', name: 'fred'}
    ])
    expect(wrapper.state('paletteInfo')).toEqual([
      {hex: '#fff', name: 'dog'},
      {hex: '#000', name: 'bob'},
      {hex: '#fff', name: 'fred'}
    ])
  });

  it('should update state when handleProject is called', () => {
    expect(wrapper.state('currentProject')).toEqual('')

    wrapper.instance().handleProject('a new project')

    expect(wrapper.state('currentProject')).toEqual('a new project')
  })
});