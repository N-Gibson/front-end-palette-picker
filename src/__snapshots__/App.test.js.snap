// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div
  className="App"
>
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/project/:id"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/palettes/:id"
    render={[Function]}
  />
</div>
`;
